packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -1099,7 +1099,36 @@ export function simplify<T>(
 			}
 			return { kind: 'union', members: members }
 		case 'struct':
-			const fields: SimplifiedStructTypePairField[] = []
+			const literalFields = new Map<string, StructTypePairField>()
+			let complexFields: SimplifiedStructTypePairField[] = []
```

Comment:
these collections live inside `simplify` and not e.g. globally, guessing this is intentional. does this mean that these collections only apply to the struct at the current level, and we won't re-use keys for structs on typeDefs above/below the current struct?

(my wording my be a bit off here)

Reply:
Correct. These are temporary collecting variables only used in the current struct case, and at the end combines into a result.